components/AudioPlayer.tsx

```diff
@@ -0,0 +1,388 @@
+          "& img": {
+            position: "absolute",
+            left: "50%",
+            top: "50%",
+            transform: "translate(-50%, -50%)",
+          },
+          "&:hover": {
+            backgroundColor: "white",
+          },
+        }}
+        onClick={togglePlayPause}
+      >
+        {isPlaying ? (
+          <img
+            style={{
+              filter:
+                "invert(20%) sepia(0%) saturate(87%) hue-rotate(231deg) brightness(95%) contrast(89%)",
+              objectFit: "cover",
+            }}
+            src="/images/pause-icon.svg"
+            alt="play"
+            width={15}
+            height={14}
+          />
+        ) : (
+          <img src="/images/play-icon.svg" alt="play" width={16} height={15} />
+        )}
+      </IconButton>
+      <Box
+        sx={{
+          margin: "5px 17px 0px 7px",
+          width: "100%",
+        }}
+      >
+        <Box
+          sx={(theme) => ({
+            "& input": {
+              "--seek-before-width": 0,
+              appearance: "none",
+              backgroundColor:
+                theme.palette.mode === "light" ? "white" : "#18191d",
+              borderRadius: "4px",
+              position: "relative",
+              width: "100%",
+              height: "4px",
+              outline: "none",
+              cursor: "pointer",
+              "&::before": {
+                content: '""',
+                height: "4px",
+                width: "var(--seek-before-width)",
+                backgroundColor: "#fe6c0a",
+                borderRadius: "4px",
+                position: "absolute",
+                zIndex: 2,
+              },
+              "&::-webkit-slider-thumb": {
+                opacity: 0,
+              },
+            },
+          })}
+        >
+          <input
+            type="range"
+            defaultValue="0"
+            ref={progressBar}
+            onChange={changeRangeOfProgressBar}
+          />
+        </Box>
+
+        <Box
+          sx={(theme) => ({
+            display: "flex",
+            justifyContent: "space-between",
+            marginTop: "2px",
+            marginLeft: "2px",
+            color: theme.palette.mode === "light" ? "#797979" : "#9f9f9f",
+            fontSize: "14px",
+          })}
+        >
+          <Box>{calculateTime(currentTime)}</Box>
+          <Box>{calculateTime(duration)}</Box>
+        </Box>
+      </Box>
+      <IconButton
+        sx={{
+          width: "20px",
+          borderColor: "transparent",
+          mr: 0.5,
+          "&:hover": {
+            background: "none",
+          },
+        }}
+        disableRipple
+        onMouseEnter={() => {
+          if (!isMobile()) {
+            setActiveVolume(true);
+          }
+        }}
+        onMouseLeave={() => {
+          if (!isMobile()) {
+            timeoutRef.current = setTimeout(() => {
+              setActiveVolume(false);
+            }, 500);
+          }
+        }}
```

Comment:
Looks like some kind of hack is going on in here

Reply:
hmm, what do you mean? You call hack like variable "isMobile" or events "onMouse"?